design/ceph/object/store.md

```diff
@@ -73,7 +74,83 @@ If `preservePoolsOnDelete` is set to 'true' the pools used to support the object
   preservePoolsOnDelete: true
 ```
 
-If there is a `zone` section in object-store configuration, then the pool section in the ceph-object-zone resource will be used to define the pools.
+#### Pools shared by multiple CephObjectStore
+
+If user want to use existing pools for metadata and data, the pools must be created before the object store is created. This will be useful if multiple objectstore can share same pools. The detail of pools need to shared in `radosNamespaces` settings in object-store CRD. And for this option to work, `preservePoolsOnDelete` must be set to 'true'. Now the object stores can consume same pool isolated with different namespaces. User can create via [Pool CRD](/Documentation/CRDs/Block-Storage/ceph-block-pool-crd.md), this is need to present before the object store is created.
+
+```yaml
+  radosNamespaces:
+    metadataPoolName: rgw-meta-pool
+    dataPoolName: rgw-data-pool
+  preservePoolsOnDelete: true
+```
+
+To create the pools that will be shared by multiple object stores, create the following CephBlockPool CRs:
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+    name: rgw-meta-pool
+  spec:
+    failureDomain: host
+    replicated:
+      size: 3
+    parameters:
+      pg_num: 8
+    application: rgw
+---
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+ metadata:
+    name: rgw-data-pool
+  spec:
+    failureDomain: osd
+      erasureCoded:
+        dataChunks: 6
+        codingChunks: 2
+    application: rgw
+```
+
+The `.rgw.root` pool must be defined before the object store is created. This pool is used for storing the root info for all object stores.
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+    name: rgw-root
+  spec:
+    name: .rgw.root
+    failureDomain: host
+    replicated:
+      size: 3
+    parameters:
+      pg_num: 8
+    application: rgw
+```
+
+The pools for this configuration will be created as below:
+
+```bash
+# ceph osd pool ls
+.rgw.root
+rgw-meta-pool.rgw.control:my-store
```

Comment:
This RADOS namespace will be inside the pool `rgw-meta-pool`, right? So is this Or is this the `my-store` RADOS namespace inside the pool `rgw-meta-pool.rgw.control`? But I thought there was only one metadata pool created.

Reply:
radosnamespace will be the store/zone name. We create pools with pool config. I got example from https://github.com/rook/rook/issues/10805#issuecomment-1340013429

Reply:
From that comment I had understood that there could be one meta pool, and then several RADOS namespaces created inside that single pool. I thought this is why your design doc only created two pools above: `rgw-meta-pool` and `rgw-data-pool` (let's ignore `.rgw.root` right now). Then all the RADOS namespaces for the metadata can be created inside the single metadata pool. There is no need for more than these two pools anymore as I understand it. The purpose of the RADOS namespace is so we can avoid creating more pools, and the object stores can also share the same pools, with different RADOS namespaces.

For example, let's say we want to create two object stores that share the same pools, where the object stores are named "one" and "two". This means we create two pools, with their RADOS namespaces in the sub-bullets: 
- `rgw-meta-pool` 
- `rgw-data-pool`

In that other link, this would translate to the rgw config for the first object store (This list is likely *not* complete, but it's just following the example in the other link):
```
"user_keys_pool": "rgw-meta-pool:one.users.keys",
"user_email_pool": "rgw-meta-pool:one.users.email",
"user_swift_pool": "rgw-meta-pool:one.users.swift",
"user_uid_pool": "rgw-meta-pool:one.users.uid",
```
And the same pools with different RADOS namespaces for the second object store:
```
"user_keys_pool": "rgw-meta-pool:two.users.keys",
"user_email_pool": "rgw-meta-pool:two.users.email",
"user_swift_pool": "rgw-meta-pool:two.users.swift",
"user_uid_pool": "rgw-meta-pool:two.users.uid",
```
Therefore, multiple RADOS namespaces would be created for each object store in those same two pools.

We can discuss in our next meeting to clarify.

Reply:
I am also confused. The output here says that the command is `ceph osd pool ls`, which lists pools. There are many more pools in this output than the 3 pools defined above. What's going on here? 

Maybe it would be more helpful to list out pools as a bulleted list, with namespaces and objects as sub-components instead of this output that I and many others probably don't understand natively.